craft_providers/multipass/multipass.py

```diff
@@ -0,0 +1,382 @@
+# Copyright 2021 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""API provider for Multipass.
+
+This implementation interfaces with multipass using the `multipass` command-line
+utility.
+"""
+
+import io
+import json
+import logging
+import pathlib
+import shlex
+import subprocess
+from typing import Any, Dict, List, Optional
+
+from craft_providers import errors
+
+from .errors import MultipassError
+
+logger = logging.getLogger(__name__)
+
+
+class Multipass:
+    """Wrapper for multipass command.
+
+    :param multipass_path: Path to multipass command to use.
+    """
+
+    def __init__(
+        self, *, multipass_path: pathlib.Path = pathlib.Path("multipass")
+    ) -> None:
+        self.multipass_path = multipass_path
+
+    def _run(  # pylint: disable=redefined-builtin
+        self,
+        command: List[str],
+        *,
+        check=True,
+        **kwargs,
+    ) -> subprocess.CompletedProcess:
+        """Execute command in instance_name, allowing output to console."""
+        command = [str(self.multipass_path), *command]
+
+        logger.debug("Executing on host: %s", shlex.join(command))
+        return subprocess.run(command, check=check, **kwargs)
```

Comment:
If `check` is not used in other place, what's the point in separating it from `kwargs`?

Reply:
check is defaulted to True. While I try to be explicit about check all of the time, it's a safe(r) default and avoids the pylint complaint of:
`craft_providers/multipass/multipass.py:56:15: W1510: Using subprocess.run without explicitly set `check` is not recommended. (subprocess-run-check)`  If there's an easy to punt that check, I'd be interested :D